networkx/classes/backends.py

```diff
@@ -131,7 +131,10 @@ def wrapper(*args, **kwds):
         if args:
             graph = args[0]
         else:
-            graph = kwds["G"]  # raise KeyError if not present
+            try:
+                graph = kwds["G"]
+            except KeyError:
+                raise TypeError(f"{name}() missing positional argument: 'G'")
```

Comment:
Great! I would also add ` from None` so you don't get chained error messages (i.e., `"During handling of the above exception, another exception occurred:"`):
```suggestion
                raise TypeError(f"{name}() missing positional argument: 'G'") from None
```

Reply:
Hmm, don't we have some places which expects graphs not as `G`, like `graph`?

Reply:
Doesn't look like many, but there are a few in this file such as `to_directed(graph)`.
https://github.com/networkx/networkx/blob/main/networkx/classes/function.py#L512

A couple functions also take multiple graphs, so we'll probably need to update the `_dispatch` decorator to handle these special cases.

Currently, all wrapped functions use `G` for the graph as the first argument. Verified via:
```python
import inspect
import networkx as nx

def _get_nx_func(dispatched_func):
    # There ought to be an easier way to get the original function...
    index = dispatched_func.__code__.co_freevars.index("func")
    cell = dispatched_func.__closure__[index]
    return cell.cell_contents

for name, dispatched_func in nx.backends._registered_algorithms.items():
    func = _get_nx_func(dispatched_func)
    sig = inspect.signature(func)
    first_arg_name = next(iter(sig.parameters))
    assert first_arg_name == "G"
```
So, I think the current PR works for the upcoming NetworkX 3.1 release.